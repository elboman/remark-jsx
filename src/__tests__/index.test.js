import unified from 'unified';
import remark from 'remark-parse';

import plugin from '../index';

describe('remark-jsx plugin', () => {
  const render = text =>
    unified()
      .use(remark)
      .use(plugin, {
        componentWhitelist: ['SomeComponent'],
      })
      .processSync(text);

  it('should work correctly', () => {
    const { contents } = render('# This is markdown\n<SomeComponent />');
    expect(contents).toMatchSnapshot();
  });
});
