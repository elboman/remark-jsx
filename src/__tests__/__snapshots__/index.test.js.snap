// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-jsx plugin should work correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "This is markdown",
        },
      ],
      "properties": Object {},
      "tagName": "h1",
      "type": "element",
    },
    Object {
      "children": Array [],
      "properties": Object {},
      "tagName": "SomeComponent",
      "type": "element",
    },
  ],
  "type": "root",
}
`;
